2-Draft-Accepted/RFCNNNN-Native-Markdown-Rendering.md

```diff
@@ -0,0 +1,199 @@
+
+## Specification
+
+This RFC proposes to use `VT100` escape sequences to render markdown content.
+`ConvertFrom-Markdown` cmdlet as part of `Microsoft.PowerShell.Utility` PowerShell module would consume a string or a file path and output a PSObject containing three properties, `Tokens`, `Html`, `VT100EncodedString`.
+The `Tokens` property has the AST for the markdown document from `markdig`.
+The `Html` property has the markdown document converted to `HTML`.
+The `VT100EncodedString` property has the markdown documented with `VT100` escape sequences.
+By default, the `Tokens`and `Html` properties will be populated.
+The `VT100EncodedString` property will be populated only when `-AsVT100EncodedString` is specified.
+
+For converting strings to VT100 coded strings, we will be writing an extension to [markdig](https://github.com/lunet-io/markdig).
+The extension will insert VT100 escape sequences as appropriate.
+
+### Specification for `ConvertFrom-Markdown`
+
+```PowerShell
+
+ConvertFrom-Markdown [-Path] <string[]> [-AsVT100EncodedString] [<CommonParameters>]
+
+ConvertFrom-Markdown [-LiteralPath] <string[]> [-AsVT100EncodedString] [<CommonParameters>]
+
+ConvertFrom-Markdown [-InputObject] <psobject> [-AsVT100EncodedString] [<CommonParameters>]
+
+```
+
+#### Parameters
+
+- **Path** : Accepts an array of file paths with markdown content.
+- **LiteralPath** : Accepts an array of literal paths with markdown content.
+- **InputObject** : Accepts an InputObject of `System.IO.FileInfo`, `string` type.
+- **AsVT100EncodedString** : When selected, the `VT100EncodedString` property is populated.
+
+There will be support for changing the colors for various elements using the `Set-MarkdownOption` cmdlet. To retrieve the current settings `Get-MarkdownOption` cmdlet can be used.
+
+### Specification for `Set-MarkdownOption`
+
+```PowerShell
+
+Set-MarkdownOption [-Header1Color <ConsoleColor>] [-Header2Color <ConsoleColor>] [-Header3Color <ConsoleColor>] [-Header4Color <ConsoleColor>] [-Header5Color <ConsoleColor>] [-Header6Color <ConsoleColor>] [-CodeBlockForegroundColor <ConsoleColor>] [-CodeBlockBackgroundColor <ConsoleColor>] [-ImageAltTextForegroundColor <ConsoleColor>] [-LinkForegroundColor <ConsoleColor>] [-ItalicsForegroundColor <ConsoleColor>] [<CommonParameters>]
+
+Set-MarkdownOption -Theme <string> [<CommonParameters>]
+
+Set-MarkdownOption -InputObject <psobject> [<CommonParameters>]
+
+```
+
+The properties can individually customize the rendering on console.
+Dark will be the default theme.
+The individual colors for the dark theme are specified in the subsequent sections.
+
+### Specification for `Get-MarkdownOption`
+
+```PowerShell
+
+Get-MarkdownOption [<CommonParameters>]
+
+```
+
+### Specification for `Export-MarkdownOption`
+
+```PowerShell
+
+Export-MarkdownOption -Path <string> [<CommonParameters>]
+
+Export-MarkdownOption -LiteralPath <string> [<CommonParameters>]
+
+```
+
+Export the current markdown settings to a file.
```

Comment:
What kind of file are we exporting/importing the options as? .psd1?

Reply:
It is a JSON file, I will update to clarify.